pac4j-core/src/main/java/org/pac4j/core/engine/DefaultLogoutLogic.java

```diff
@@ -0,0 +1,152 @@
+ * or to the <code>defaultUrl</code> if it is defined or as a blank page otherwise.</p>
+ *
+ *  <p>If the <code>centralLogout</code> property is <code>true</code>, the user is redirected to the identity provider
+ * for a central logout and then optionally to the post logout redirection URL (if it's supported by the identity provider and if it's an absolute URL).
+ * If no central logout is defined, the post logout action is performed directly.</p>
+ *
+ * @author Jerome Leleu
+ * @since 1.9.0
+ */
+public class DefaultLogoutLogic<R, C extends WebContext> extends ProfileManagerFactoryAware<C> implements LogoutLogic<R, C> {
+
+    protected Logger logger = LoggerFactory.getLogger(getClass());
+
+    private boolean killSession;
+
+    @Override
+    public R perform(final C context, final Config config, final HttpActionAdapter<R, C> httpActionAdapter,
+                     final String defaultUrl, final String inputLogoutUrlPattern, final Boolean inputLocalLogout,
+                     final Boolean inputCentralLogout) {
+
+        logger.debug("=== APP LOGOUT ===");
+
+        // default values
+        final String logoutUrlPattern;
+        if (inputLogoutUrlPattern == null) {
+            logoutUrlPattern = Pac4jConstants.DEFAULT_LOGOUT_URL_PATTERN_VALUE;
+        } else {
+            logoutUrlPattern = inputLogoutUrlPattern;
+        }
+        final boolean localLogout;
+        if (inputLocalLogout == null) {
+            localLogout = true;
+        } else {
+            localLogout = inputLocalLogout;
+        }
+        final boolean centralLogout;
+        if (inputCentralLogout == null) {
+            centralLogout = false;
+        } else {
+            centralLogout = inputCentralLogout;
+        }
+
+        // checks
+        assertNotNull("context", context);
+        assertNotNull("config", config);
+        assertNotNull("httpActionAdapter", httpActionAdapter);
+        assertNotBlank(Pac4jConstants.LOGOUT_URL_PATTERN, logoutUrlPattern);
+        final Clients configClients = config.getClients();
+        assertNotNull("configClients", configClients);
+
+        // logic
+        final ProfileManager manager = getProfileManager(context);
+        final List<CommonProfile> profiles = manager.getAll(true);
+
+        // compute redirection URL
+        final String url = context.getRequestParameter(Pac4jConstants.URL);
+        String redirectUrl = defaultUrl;
+        if (url != null && Pattern.matches(logoutUrlPattern, url)) {
+            redirectUrl = url;
+        }
+        logger.debug("redirectUrl: {}", redirectUrl);
+        HttpAction action;
+        if (redirectUrl != null) {
+            action = HttpAction.redirect("redirect", context, redirectUrl);
+        } else {
+            action = HttpAction.ok("ok", context);
+        }
+
+        // local logout if requested or multiple profiles
+        if (localLogout || profiles.size() > 1) {
```

Comment:
I changed the logic here as well.

The `localLogout` and the  `centralLogout` may be disabled/enabled.